### PR TITLE
[Refactor] [Step 1/2]Change the name from master to leader

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Change the master related code to leader.
Split the task into two steps: 
1. change the name from master to leader in the internal code paths.
2. change the name from master to leader in the interface.

The advantage of this is that we only need to rollback the step 2 if If there are compatibility problems.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 123 (27.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/CancelStreamLoad.java|0|1|00.00%|[51]|
|:large_blue_circle:|com/starrocks/http/rest/ShowProcAction.java|0|4|00.00%|[74, 78, 83, 91]|
|:large_blue_circle:|com/starrocks/http/rest/ShowMetaInfoAction.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/http/rest/GetStreamLoadState.java|0|1|00.00%|[51]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJob.java|0|1|00.00%|[1101]|
|:large_blue_circle:|com/starrocks/http/meta/GlobalDictMetaService.java|0|1|00.00%|[43]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|0|1|00.00%|[279]|
|:large_blue_circle:|com/starrocks/http/rest/GetLoadInfoAction.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|4|00.00%|[42, 109, 110, 111]|
|:large_blue_circle:|com/starrocks/http/action/HaAction.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|4|00.00%|[89, 92, 95, 102]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|2|00.00%|[126, 130]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|0|2|00.00%|[123, 152]|
|:large_blue_circle:|com/starrocks/metric/PrometheusMetricVisitor.java|0|1|00.00%|[220]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|2|00.00%|[375, 376]|
|:large_blue_circle:|com/starrocks/http/meta/MetaService.java|0|1|00.00%|[180]|
|:large_blue_circle:|com/starrocks/http/meta/ColocateMetaService.java|0|1|00.00%|[96]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|2|11|18.18%|[797, 802, 871, 951, 1063, 1234, 1239, 1244, 1249]|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|1|5|20.00%|[143, 167, 224, 461]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|9|22.22%|[245, 258, 261, 270, 273, 577, 579]|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|1|3|33.33%|[76, 104]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|3|9|33.33%|[803, 1083, 1084, 1093, 1149, 1214]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|3|33.33%|[504, 505]|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|10|26|38.46%|[905, 912, 919, 923, 924, 925, 1025, 1026, 1027, 1028, 1029, 1036, 1038, 1039, 1041, 1042]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|9|22|40.91%|[1110, 1375, 1629, 1647, 1658, 1662, 1672, 1678, 1736, 2539, 2543, 2547, 2571]|
|:large_blue_circle:|com/starrocks/ha/LeaderInfo.java|1|2|50.00%|[33]|
|:large_blue_circle:|com/starrocks/ha/BDBStateChangeListener.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/ha/FrontendNodeType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/LeaderDaemon.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public class FrontendsProcNode implements ProcNodeInterface {
 
     public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
             .add("Name").add("IP").add("EditLogPort").add("HttpPort").add("QueryPort").add("RpcPort")
-            .add("Role").add("IsMaster").add("ClusterId").add("Join").add("Alive").add("ReplayedJournalId")
+            .add("Role").add("ClusterId").add("Join").add("Alive").add("ReplayedJournalId")
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java:49]: I think you should keep isMaster in this PR, then this PR don't change user interface. we can merge this PR quickly.
And you make this change in another PR.

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterHandler.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
 import com.starrocks.common.MetaNotFoundException;
 import com.starrocks.common.ThreadPoolManager;
 import com.starrocks.common.UserException;
-import com.starrocks.common.util.MasterDaemon;
+import com.starrocks.common.util.LeaderDaemon;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterHandler.java:36]: Add a more clear messages in the commit message.

Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/alter/AlterHandler.java:36]: If not be compatible, we should remove the IsMaster in the printed result.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java]: done

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 33 / 125 (26.40%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/CancelStreamLoad.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/http/rest/ShowProcAction.java|0|4|00.00%|[74, 78, 83, 91]|
|:large_blue_circle:|com/starrocks/http/rest/ShowMetaInfoAction.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/http/rest/GetStreamLoadState.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJob.java|0|1|00.00%|[1101]|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|0|5|00.00%|[75, 76, 77, 102, 104]|
|:large_blue_circle:|com/starrocks/http/meta/GlobalDictMetaService.java|0|1|00.00%|[43]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|0|1|00.00%|[279]|
|:large_blue_circle:|com/starrocks/http/rest/GetLoadInfoAction.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|4|00.00%|[43, 110, 111, 112]|
|:large_blue_circle:|com/starrocks/http/action/HaAction.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|4|00.00%|[89, 92, 95, 102]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|2|00.00%|[125, 129]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|0|2|00.00%|[123, 152]|
|:large_blue_circle:|com/starrocks/metric/PrometheusMetricVisitor.java|0|1|00.00%|[220]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|2|00.00%|[375, 376]|
|:large_blue_circle:|com/starrocks/http/meta/MetaService.java|0|1|00.00%|[180]|
|:large_blue_circle:|com/starrocks/http/meta/ColocateMetaService.java|0|1|00.00%|[95]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|2|11|18.18%|[797, 802, 871, 951, 1063, 1234, 1239, 1244, 1249]|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|1|5|20.00%|[143, 167, 224, 461]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|9|22.22%|[245, 258, 261, 270, 273, 577, 579]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|3|9|33.33%|[803, 1083, 1084, 1093, 1149, 1214]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|3|33.33%|[504, 505]|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|10|26|38.46%|[905, 912, 919, 923, 924, 925, 1025, 1026, 1027, 1028, 1029, 1036, 1038, 1039, 1041, 1042]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|9|22|40.91%|[1110, 1375, 1629, 1647, 1658, 1662, 1672, 1678, 1736, 2539, 2543, 2547, 2571]|
|:large_blue_circle:|com/starrocks/ha/LeaderInfo.java|1|2|50.00%|[33]|
|:large_blue_circle:|com/starrocks/ha/BDBStateChangeListener.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/ha/FrontendNodeType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/LeaderDaemon.java|2|2|100.00%|[]|

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test
